db/models/news.model.js

```diff
@@ -7,23 +7,37 @@ exports.selectTopics = () => {
         return response.rows
     })
     .catch((error) => {
-        console.error("Error selecting topics:", error);
         throw error;
     });
 }
 
 exports.selectArticleByID = (article_id) => {
-    if (isNaN(article_id)) {
-        return Promise.reject({ status: 400, message: "Bad Request - This is not a valid Article ID" });
-    }
-    else {
     return db.query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id])
     .then((response) => {
-        return response.rows[0]
+        const article = response.rows[0];
+        if (!article) {
+          return Promise.reject({
+            status: 404,
+            msg: `No article found for article_id: ${article_id}`,
+        })
+        }
+    return article
     })
     .catch((error) => {
-        console.error("Error selecting topics:", error);
         throw error;
     });
 }
+
+
+exports.selectArticles = () => {
+    return db.query(`SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, 
+    articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM articles 
+    JOIN comments ON comments.article_id = articles.article_id
+    GROUP BY articles.article_id ORDER BY articles.created_at DESC;`)
+    .then((response) => {
+        return response.rows
+    })
+    .catch((error) => {
+        throw error;
```

Comment:
please us a promise rejection instead of throwing and error! this is better practice when working with promise. chain.